fearless/Modules/Staking/StakingAmount/StakingAmountProtocols.swift

```diff
@@ -57,5 +57,5 @@ protocol StakingAmountWireframeProtocol: AlertPresentable, ErrorPresentable,
 }
 
 protocol StakingAmountViewFactoryProtocol: class {
-	static func createView() -> StakingAmountViewProtocol?
+    static func createView(with amount: Decimal?) -> StakingAmountViewProtocol?
```

Comment:
It is good to pass amount value but it is not recalculated when fee information is received. If amount is 100% of available balance then user will see insufficient funds alert which is confusing. I'd suggest checking the amount against (available balance - fee) whenever new fee is available
<img width="553" alt="image" src="https://user-images.githubusercontent.com/3176149/110382146-8bc91800-806b-11eb-90f5-8040e53d88f0.png">
<img width="553" alt="image" src="https://user-images.githubusercontent.com/3176149/110382194-9be0f780-806b-11eb-98af-3d69e59bf1fa.png">

Reply:
For such a use case we need fee information on the main screen. Otherwise it will be strange that we display one amount on the main screen and another amount later.

Reply:
I meant checking amount on the next screen after main. If Amount + Fee is greater than available amount, we could just decrease it to be equal to max available amount or notify user with an alert that might look like "Amount you entered before is insufficient because of fee. Please, recalculate". Let's leave it as it is again and discuss later